app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -40,18 +43,33 @@ class PatientsScreenController @Inject constructor(
   }
 
   private fun refreshApprovalStatusOnStart(events: Observable<UiEvent>): Observable<UiChange> {
-    // Skipping the first resume because it'll be a duplicate of screen-created.
     val screenCreate = events.ofType<ScreenCreated>()
-    val screenResumes = events.ofType<TheActivityLifecycle.Resumed>().skip(1)
+    val screenResumes = events.ofType<TheActivityLifecycle.Resumed>()
 
     return Observable.merge(screenCreate, screenResumes)
-        .flatMap {
+        // Depending upon the entry point of Patients screen, both screen-create
+        // and screen-resume events may or may not happen at the same time.
+        // So duplicate triggers are dropped by restricting flatMap's
+        // concurrency to 1.
+        .toFlowable(BackpressureStrategy.LATEST)
+        .flatMapMaybe({
           userSession.loggedInUser()
-              .take(1)
+              .firstOrError()
               .filter { (user) -> user!!.status == WAITING_FOR_APPROVAL }
-              .flatMapCompletable { userSession.refreshLoggedInUser() }
-              .andThen(Observable.never<UiChange>())
-        }
+              .doOnSuccess {
+                // The refresh call should not get canceled when the app is closed.
+                // That is, when this controller gets disposed by the screen.
+                userSession.refreshLoggedInUser()
+                    .doOnComplete {
+                      approvalStatusUpdatedAtPref.set(Instant.now())
+                    }
+                    .onErrorComplete()
```

Comment:
Good point. I don't think this will affect the existing logic because we stop refreshing the status as soon as the status changes to `approved`. And this timestamp is only used after the status changes to `approved.`

It's used in `PatientsScreenController#90`. Do you think this looks alright?

Reply:
Seems fine to me then. 👍